eml/screens/explore/Explore.jsx

```diff
@@ -1,87 +1,203 @@
+*/
+  function shouldUpdate(courses1, courses2) {
+    // If both arrays are empty, they are equal, but should still update
+    if (courses1.length === 0 && courses2.length === 0) {
+      return true;
+    }
+
+    // If the lengths are different, they are not equal
+    if (courses1.length !== courses2.length) {
+      return true;
+    }
+
+    // If the IDs are different, they are not equal
+    for (let i = 0; i < courses1.length; i++) {
+      if (courses1[i].id !== courses2[i].id) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+  * Asynchronous function that loads the subscribed courses from storage and updates the state.
+  * @returns {void}
+  */
+  async function loadSubscriptions() {
+    const subData = await StorageService.getSubCourseList();
+    if (shouldUpdate(subCourses, subData)) {
+      if (subData.length !== 0 && Array.isArray(subData)) {
+        setSubCourses(subData);
+      }
+      else {
+        setSubCourses([]);
+      }
+    }
+  }
+
+  /**
+  * Asynchronous function that loads the courses from storage and updates the state.
+  * @returns {void}
+  */
+  async function loadCourses() {
+    const courseData = await StorageService.getCourseList();
+    if (shouldUpdate(courses, courseData)) {
+      if (courseData.length !== 0 && Array.isArray(courseData)) {
+        setCourses(courseData);
+      }
+      else {
+        setCourses([]);
+      }
+    }
+  }
+
+  // When refreshing the loadCourse and load subscription function is called
+  const onRefresh = () => {
+    setRefreshing(true);
+    loadSubscriptions();
+    loadCourses();
+    // Fetch subscriptions for filtered courses and set them in state
+    fetchSubscriptionsForFilteredCourses().then((results) => {
+      setIsSubscribed(results);
+    });
+    setRefreshing(false);
+  };
+
+  // Function to check if user is subscribed to a specific course
+  async function fetchCourseSubscription(course) {
+    const result = await StorageService.checkSubscriptions(course.courseId);
+    return result;
+  }
+
+  // Function to check if user is subscribed to all filtered courses
+  async function fetchSubscriptionsForFilteredCourses() {
+    const results = await Promise.all(
+      filteredCourses.map((course) => fetchCourseSubscription(course))
+    );
+    return results;
+  }
+
+  useEffect(() => {
+    // this makes sure loadcourses is called when the screen is focused
+    const update = navigation.addListener('focus', () => {
+      loadCourses();
+      loadSubscriptions();
+    });
+    // Fetch subscriptions for filtered courses and set them in state
+    fetchSubscriptionsForFilteredCourses().then((results) => {
+      setIsSubscribed(results);
+    });
+    return update;
+
+  }, [navigation, subCourses]);
+
+  ///---------------------------------------------///
+
+  // Function to dermine category
+  function determineCategory(category) {
```

Comment:
Why is this function defined again here, when it is defined in utilityfunctions.js

Reply:
Because I am stupid, I will fix it lol